application/F3DOptionsParser.cxx

```diff
@@ -693,49 +724,55 @@ bool ConfigurationOptions::InitializeDictionaryFromConfigFile(const std::string&
     }
   }
 
-  fs::path configPath;
+  std::vector<fs::path> actualConfigFilePaths;
+
+  std::vector<fs::path> configPaths;
+
   if (!configSearch.empty())
   {
-    configPath = F3DConfigFileTools::GetConfigPath(configSearch);
+    for (const auto& path : F3DConfigFileTools::GetConfigPaths(configSearch))
+    {
+      configPaths.emplace_back(path);
+    }
   }
   else
   {
-    configPath = config;
+    configPaths.emplace_back(config);
   }
 
-  if (configPath.empty())
+  for (auto configPath : configPaths)
   {
-    if (!config.empty())
+    // Recover an absolute canonical path to config file
+    try
     {
-      f3d::log::error("Configuration file for \"", config, "\" could not been found");
+      configPath = fs::canonical(fs::path(configPath)).string();
+    }
+    catch (const fs::filesystem_error&)
+    {
+      f3d::log::error("Configuration file does not exist: ", configPath.string());
+      return false;
     }
-    return false;
-  }
-
-  // Recover an absolute canonical path to config file
-  try
-  {
-    configPath = fs::canonical(fs::path(configPath)).string();
-  }
-  catch (const fs::filesystem_error&)
-  {
-    f3d::log::error("Configuration file does not exist: ", configPath.string());
-    return false;
-  }
-  f3d::log::debug("Using config file ", configPath.string());
 
-  // Recover all config files if needed
-  std::set<fs::path> actualConfigFilePaths;
-  if (fs::is_directory(configPath))
-  {
-    for (auto& entry : std::filesystem::directory_iterator(configPath))
+    // Recover all config files if needed
+    if (fs::is_directory(configPath))
     {
-      actualConfigFilePaths.emplace(entry);
+      f3d::log::debug("Using config directory ", configPath.string());
+      for (auto& entry : std::filesystem::directory_iterator(configPath))
+      {
+        actualConfigFilePaths.emplace_back(entry);
+      }
+    }
+    else
+    {
+      f3d::log::debug("Using config file ", configPath.string());
+      actualConfigFilePaths.emplace_back(configPath);
     }
   }
-  else
+
+  if (actualConfigFilePaths.empty())
   {
-    actualConfigFilePaths.emplace(configPath);
+    f3d::log::error("Configuration file for \"", config, "\" could not be found");
```

Comment:
One think I dont like with this is that when not specifying any `--config`, then `config` is empty resulting into:

`f3d::log::error("Configuration file for \"", config, "\" could not be found");`

Maybe this `config` var should be initialized to `config` much higher up.

Reply:
TBH I didn't try to fully understand the `--config` logic, as per other comment I think there would be some refactoring to do to separate the config file search and parsing. 
Unless it's a blocker I'd rather leave that for someone else to do.

Reply:
not a blocker

Reply:
did the minimum of avoiding the bad log, proper fix of the logic would need a wider refactoring as already discussed